frontend/packages/console-app/src/components/detect-language/DetectLanguage.tsx

```diff
@@ -0,0 +1,10 @@
+import * as React from 'react';
+import { usePreferredLanguage, useLanguage } from '../user-preferences/language';
+
+const DetectLanguage: React.MemoExoticComponent<() => any> = React.memo(() => {
+  const [preferredLanguage, , preferredLanguageLoaded] = usePreferredLanguage();
```

Comment:
Did we lose the query parameter detection for language?

Reply:
I think `i18next-browser-languageDetector` handles that. The old language preference modal didn't seem to detect language from query params as well.
The `Detect Language` just populates the local storage on first load if there is a preferred language in configmap but the same does not reflect in local storage.
This will be useful if the user changes browser after setting a language preference.